### DISCUSSION TITLE
error compiling example sketch streams-i2s-a2dp

### DISCUSSION THREAD

**Discussion by rkb68** [pschatzmann/arduino-audio-tools#729] [Q&A] [answered]
Hello,

Using arduino-audio-tools 04f44f5 (commit of 2023-03-29)

Upon compiling in Arduino (1.8.19), I get this error:

In file included from /.../Arduino/libraries/arduino-audio-tools-main/src/AudioI2S/I2SStream.h:11:0,
                 from /.../Arduino/libraries/arduino-audio-tools-main/src/AudioTools.h:60,
                 from /.../Arduino/streams-i2s-a2dp/streams-i2s-a2dp.ino:10:
/.../Arduino/libraries/arduino-audio-tools-main/src/AudioI2S/I2SESP32.h:183:42: error: missing binary operator before token "("
#if ESP_IDF_VERSION > ESP_IDF_VERSION_VAL(4, 4, 0)

The is with the Sparkfun ESP32 library v 1.0.1 and the Espressif ESP32 library v 2.0.7

Has anybody else encountered this?
Thanks, rkb

**Comment by pschatzmann** [comment, marked as answer]
What is Sparkfun ESP32 library v 1.0.1 ?
Maybe it destroys some functionality of the generic ESP32 library

I tested in Arduino (1.8.19)  with ESP32 library v 2.0.7 and did not have any issues

The ESP_IDF_VERSION_VAL macro is part of #include "esp_idf_version.h" and can be used to check for the IDF version

I suggest that you try to reinstall ESP32 v 2.0.7

**Reply by rkb68** [reply]
Thank you.   The Sparkfun library I referred to is in fact the board support package for many of their ESP32 products.  It usually coexists peacefully with the Espressif package, but evidently not all the time.

**Reply by rkb68** [reply]
Success!

Thanks again; did not have HW available when replying earlier; can now compile without error by bypassing the Sparkfun library.

In case anybody else has this problem, I was able to select the board (Sparkfun ESP32 Thing Plus C) not from the "Sparkfun ESP32 Arduino" folder, but instead from the "ESP32 Arduino" folder, and everything works just fine.
